docs/local_assembly.tex

```diff
@@ -0,0 +1,121 @@
+\documentclass[nofootinbib,amssymb,amsmath]{revtex4}
+\usepackage{mathtools}
+\usepackage{amsthm}
+\usepackage{algorithm}
+\usepackage{algpseudocode}
+\usepackage{lmodern}
+\usepackage{graphicx}
+\usepackage{color}
+
+%Put an averaged random variable between brackets
+\newcommand{\ave}[1]{\left\langle #1 \right\rangle}
+\newcommand{\HC}{\texttt{HaplotypeCaller}}
+\newcommand{\Mutect}{\texttt{Mutect}}
+\newcommand{\code}[1]{\texttt{#1}}
+
+\newtheorem{lemma}{Lemma}
+\newtheorem{corollary}{Corollary}
+
+\def\SL#1{{\color [rgb]{0,0,0.8} [SL: #1]}}
+\def\DB#1{{\color [rgb]{0,0.8,0} [DB: #1]}}
+
+\begin{document}
+
+\title{Local Assembly in HaplotypeCaller and Mutect}
+\author{David Benjamin\footnote{The author took no part in development of the methods described below -- credit belongs to several others on the GATK team. }}
+\email{davidben@broadinstitute.org}
+\affiliation{Broad Institute, 75 Ames Street, Cambridge, MA 02142}
+
+\date{\today}
+
+\begin{abstract}
+The GATK tools \HC~and \Mutect~assemble reads aligned within a window of several hundred base pairs into a de Bruijn graph of local variation.  Local haplotypes correspond to paths in this de Bruijn graph, and the downstream likelihood calculations in both tools involve aligning reads to these haplotypes.  Here we describe how this de Bruijn graph is generated.
```

Comment:
You might be overestimating me when you say "I know you know that."  I'm trying to figure out exactly what makes our read threading graph (before pruning and clean-up, of course) not a de Bruijn graph.  The accepted definition in the literature seems to me to be:

* Vertices are kmers
* Two vertices get a directed edge whenever they appear consecutively in the same read.

Our graph differs in two ways that I can think of:

* We keep track of the multiplicity of edges.
* We thread the reference haplotype and GGA sequences as if they were reads.

I can certainly accept being more precise and calling our graph a weighted de Bruijn graph with a reference pseudo-read, but I fear that you have another, more fundamental difference in mind.

Reply:
Kmer <-> vertex is not a bijective relationship, a kmer may have more than one vertex (non-unique kmers at the very least). Also partly as consequence of that we don't have many cycles that otherwise are present in the deBruijn graph.

Reply:
Also due to the way we construct the graph... this may well be different depending on what order we process the sequences.... scary huh!? The union of the individual sequences deBruijn's would be the same graph regardless of the order the sequences are processed... I think.

Reply:
Addendum: another difference based on verifying your points is that we create a new vertex for non-unique kmers, even those that already appear on the graph.  Is there anything else?

Reply:
> Kmer <-> vertex is not a bijective relationship, a kmer may have more than one vertex (non-unique kmers at the very least)

Okay, I get how this is fundamental.  I think I will change the discussion to explicitly highlight the differences.  That way people who are familiar with de Bruijn graphs can find comfort in the fact that most of their knowledge carries over to these docs, but will not be led to believe that everything is the same.

Reply:
I changed the docs.